### PR TITLE
fix unhandeled error 412 of sso

### PR DESCRIPTION
**Pull request by YitongFeng-git**
1. catch 412 error
2. unify quotes

### CODE REVIEW

File: bot-msgext/js/index.js [hunk 1]
```diff
@@ -68,9 +68,13 @@ server.listen(process.env.port || process.env.PORT || 3978, function () {
 });
 
 // Listen for incoming requests.
-server.post('/api/messages', (req, res) => {
-    adapter.processActivity(req, res, async (context) => {
+server.post('/api/messages', async (req, res) => {
+    await adapter.processActivity(req, res, async (context) => {
         await bot.run(context);
+    }).catch(err => {
```
Comment (henzhang-ms) [inline bot-msgext/js/index.js:74]: Nit: please ensure whether use try-cache or cache function

File: bot-msgext/js/index.js [hunk 2]
```diff
@@ -68,9 +68,13 @@ server.listen(process.env.port || process.env.PORT || 3978, function () {
...
+        if (!err.message.includes('412')) {
```
Comment (henzhang-ms) [inline bot-msgext/js/index.js:75]: better to add comments for explanation.

File: bot-msgext/ts/index.ts [hunk 3]
```diff
@@ -74,9 +74,13 @@ server.listen(process.env.port || process.env.PORT || 3978, () => {
 });
 
 // Listen for incoming requests.
-server.post("/api/messages", (req, res) => {
-    adapter.processActivity(req, res, async (context) => {
+server.post('/api/messages', async (req, res) => {
```
Comment (henzhang-ms) [inline bot-msgext/ts/index.ts:77]: it is ts, maybe we should use `"`

File: bot/ts/index.ts [hunk 4]
```diff
@@ -74,9 +74,13 @@ server.listen(process.env.port || process.env.PORT || 3978, () => {
 });
 
 // Listen for incoming requests.
-server.post("/api/messages", (req, res) => {
-    adapter.processActivity(req, res, async (context) => {
+server.post('/api/messages', async (req, res) => {
```
Comment (henzhang-ms) [inline bot/ts/index.ts:77]: `"`

**Review by henzhang-ms — APPROVED** [review]

File: bot-msgext/ts/index.ts [hunk 5]
```diff
@@ -74,9 +74,13 @@ server.listen(process.env.port || process.env.PORT || 3978, () => {
 });
 
 // Listen for incoming requests.
-server.post("/api/messages", (req, res) => {
-    adapter.processActivity(req, res, async (context) => {
+server.post("/api/messages", async (req, res) => {
+    await adapter.processActivity(req, res, async (context) => {
         await bot.run(context);
+    }).catch(err => {
+        if (!err.message.includes("412")) {
```
Comment (IvanJobs) [inline bot-msgext/ts/index.ts:81]: Nit: risky code to check 412. If no other specific way to do the checking, ignore this comment.

**Review by IvanJobs — APPROVED** [review]

File: bot-msgext/js/index.js (hunk 2 above)
Comment (YitongFeng-git) [reply bot-msgext/js/index.js:75]: added

File: bot-msgext/ts/index.ts (hunk 5 above)
Comment (YitongFeng-git) [reply bot-msgext/ts/index.ts:81]: Yes, the error message is "412 Error"

**Pull request merged**